Library/PKPaymentAuthorizationViewController.swift

```diff
@@ -27,13 +27,14 @@ extension PKPaymentAuthorizationViewController {
 
     if countryCode == "US" && project.country != Project.Country.us ||
       countryCode != "US" {
-      return PKPaymentAuthorizationViewController.supportedNetworks.filter { $0 != .discover }
+      return PKPaymentAuthorizationViewController.supportedNetworks
+        .filter { $0 != .discover && $0 != .chinaUnionPay }
```

Comment:
According to this PR https://github.com/kickstarter/kickstarter/blob/master/browser/javascripts/stripe-elements/credit-cards/utils.ts , UnionPay should not be supported for non-US projects.

Reply:
This is starting to look really confusing since we do have 

`static func supportedNetworks(for project: Project) ...` 

and then

`static func supportedNetworks`

I wonder if it would be clearer with something like this:

```
import KsApi
import PassKit

extension PKPaymentAuthorizationViewController {

  public static var merchantIdentifier: String {
    return "merchant.com.kickstarter"
  }

  public static func applePayCapable() -> Bool {
    return PKPaymentAuthorizationViewController.canMakePayments(usingNetworks: PaymentNetworks.all)
  }

  public static func applePayDevice() -> Bool {
    return PKPaymentAuthorizationViewController.canMakePayments()
  }

  public static func applePayCapable(for project: Project) -> Bool {
    return PKPaymentAuthorizationViewController.canMakePayments(
      usingNetworks: PKPaymentAuthorizationViewController.supportedNetworks(for: project)
    )
  }

  public static func supportedNetworks(for project: Project) -> [PKPaymentNetwork] {
    let countryCode = AppEnvironment.current.countryCode

    if (countryCode == "US" && project.country != Project.Country.us) || countryCode != "US" {
      return PaymentNetworks.US.supported
    }

    return PaymentNetworks.all
  }
}

// MARK: - Payment networks

// swiftlint:disable private_over_fileprivate
fileprivate struct PaymentNetworks {
  fileprivate static var all: [PKPaymentNetwork] {
    return [.amex, .masterCard, .visa, .discover, .chinaUnionPay]
  }

  fileprivate struct US {
    fileprivate static var unsupported: [PKPaymentNetwork] {
      return [.chinaUnionPay, .discover]
    }

    fileprivate static var supported: [PKPaymentNetwork] {
      return PaymentNetworks.all.filter { US.unsupported.contains($0) }
    }
  }
}
// swiftlint:enable private_over_fileprivate
```

What do you think Nino? (this is just a suggestion, I'd be OK with this approach for now)

Reply:
Also I would consider putting `countryCode == "US" && project.country != Project.Country.us` in parenthesis for better readability.

Reply:
This is so much better! Thanks, @dusi !